### DISCUSSION TITLE
How to restart geocoding?

### DISCUSSION THREAD

**Discussion by larsrh** [Freika/dawarich#346] [Q&A]
Before I updated to 0.15.7 just now, I had a queue full of points for reverse geocoding (according to Sidekiq, about ~1M). After the update, the queue is gone. (All points are still there and I have a database backup etc.)

How can I restart geocoding?

**Comment by Freika** [comment]
Settings -> Background Jobs -> "Start Reverse Geocoding"

**Reply by larsrh** [reply]
Thanks @Freika. I did that now, but for some reason there are no further jobs in the queue (I waited for a few hours).

**Reply by Freika** [reply]
@larsrh can you see any errors in the logs?

**Reply by larsrh** [reply]
I just tried again. I see this in the log:

```
19:24:08 web.1  | I, [2024-10-22T19:24:08.144854 #127]  INFO -- : Enqueued EnqueueBackgroundJob (Job ID: 5bb01bda-e205-47f6-9e30-64f9b923c094) to Sidekiq(reverse_geocoding) with arguments: "continue_reverse_geocoding", 1
19:24:08 web.1  | I, [2024-10-22T19:24:08.145456 #127]  INFO -- : ↳ app/controllers/settings/background_jobs_controller.rb:12:in `create'
```

There's now a job in Sidekiq, but after that I see nothing relevant to reverse geocoding. Similar for my last attempt. Is there something specific I should look out for? (The only errors are caused by bots trying to hit undefined paths.)

**Reply by Freika** [reply]
Is your container `dawarich_sidekiq` running? Do you see any logs with errors there?

**Reply by larsrh** [reply]
I found a smoking gun ...

```
2024-10-22T19:24:08.145Z pid=103 tid=hl3 class=EnqueueBackgroundJob jid=0c5d9a23992e1b83ce6368ca INFO: start
I, [2024-10-22T19:24:08.859533 #103]  INFO -- : Performing EnqueueBackgroundJob (Job ID: 5bb01bda-e205-47f6-9e30-64f9b923c094) from Sidekiq(reverse_geocoding) enqueued at 2024-10-22T19:24:08.143903644Z with arguments: "continue_reverse_geocoding", 1
D, [2024-10-22T19:24:09.300350 #103] DEBUG -- :   ESC[1mESC[36mUser Load (2.3ms)ESC[0m  ESC[1mESC[34mSELECT "users".* FROM "users" WHERE "users"."id" = $1 LIMIT $2ESC[0m  [["id", 1], ["LIMIT", 1]]
D, [2024-10-22T19:24:09.301386 #103] DEBUG -- :   ↳ app/services/jobs/create.rb:10:in `initialize'
D, [2024-10-22T19:26:06.013989 #103] DEBUG -- :   ESC[1mESC[36mPoint Load (116672.7ms)ESC[0m  ESC[1mESC[34mSELECT "points".* FROM "points" WHERE "points"."user_id" = $1 AND "points"."geodata" = $2ESC[0m  [["user_id", 1], ["geodata", "{}"]]
D, [2024-10-22T19:26:06.015782 #103] DEBUG -- :   ↳ app/services/jobs/create.rb:24:in `call'
2024-10-22T19:28:25.411Z pid=103 tid=hir WARN: Your Redis network connection is performing extremely poorly.
Last RTT readings were [277535, 163421, 250515, 245292, 189124], ideally these should be < 1000.
Ensure Redis is running in the same AZ or datacenter as Sidekiq.
If these values are close to 100,000, that means your Sidekiq process may be
CPU-saturated; reduce your concurrency and/or see https://github.com/sidekiq/sidekiq/discussions/5039

... a few more of those ...

Killed
Environment: development
dawarich_db (172.23.0.3:5432) open
Latest version already installed. Done.
Successfully installed bundler-2.5.9
1 gem installed
Database dawarich_development already exists, skipping creation...
PostgreSQL is ready. Running database migrations...
```

**Reply by larsrh** [reply]
Interestingly, this only appears to happen when I try to continue reverse geocoding for the existing points. New points get happily decoded.

**Reply by Freika** [reply]
@larsrh what if you hit "Reverse geocode from scratch" button in the Background Jobs section?

And also, how does your docker-compose file look like?

**Reply by larsrh** [reply]
Sorry for the radio silence on this, I was travelling. My next step will be to migrate this from my poor VPS to a beefier home server. I'll report back, but I suspect the Sidekiq job will stop crashing there.
